### PR TITLE
Feature/geocoder building

### LINKED ISSUE #279: Return geo-point for each building in our API (for portal)

### PR DESCRIPTION
**Pull request by yorickdewid**
**What issue does this PR address?**
Fixes #279 

**Does this PR introduce a breaking change?**
Nope

### CODE REVIEW

File: src/FunderMaps.Core/Entities/Address.cs [hunk 1]
```diff
@@ -59,6 +59,24 @@ public Address()
...
+        ///     Building identifier.
+        /// </summary>
+        [Geocoder]
+        public string BuildingId { get; set; }
+
+        // TODO: Obsolete
+        /// <summary>
+        ///     Building identifier.
+        /// </summary>
+        public Building BuildingNavigation { get; set; }
```
Comment (tabeckers) [inline src/FunderMaps.Core/Entities/Address.cs:78]: Do we want an entity within an entity here?

File: src/FunderMaps.Core/Entities/Building.cs [hunk 2]
```diff
@@ -37,27 +38,33 @@ public Building()
...
         ///     External data source.
         /// </summary>
         [Required]
-        public string ExternalSource { get; set; }
+        public ExternalDataSource ExternalSource { get; set; }
+
+        /// <summary>
+        ///     Building type.
+        /// </summary>
+        public BuildingType? buildingType { get; set; }
```
Comment (tabeckers) [inline src/FunderMaps.Core/Entities/Building.cs:62]: Building type hasn't been fully implemented in the database, the format might change later. This is fine for now, just adding this as a note.

File: src/FunderMaps.Core/UseCases/GeocoderUseCase.cs [hunk 3]
```diff
@@ -42,5 +44,20 @@ public virtual async ValueTask<Address> GetAsync(string id)
...
+        ///     Retrieve all addresses matching search query.
+        /// </summary>
+        /// <param name="query">Search query.</param>
+        /// <param name="navigation">Recordset nagivation.</param>
+        public async virtual IAsyncEnumerable<Address> GetAllBySuggestionWithBuildingAsync(string query, INavigation navigation)
+        {
+            await foreach (var address in GetAllBySuggestionAsync(query, navigation))
+            {
+                // FUTURE: This is a massive performance hit.
+                address.BuildingNavigation = await _buildingRepository.GetByIdAsync(address.BuildingId);
```
Comment (tabeckers) [inline src/FunderMaps.Core/UseCases/GeocoderUseCase.cs:58]: Do we really need this? Isn't extending `Address` with a `building_geometry` field or something similar a more effective method of achieving this result? We can then return this from a database view.

File: src/FunderMaps.Core/Entities/Building.cs [hunk 4]
```diff
@@ -37,27 +38,33 @@ public Building()
         public bool IsActive { get; set; } = true;
 
         /// <summary>
-        ///     Address identifier.
+        ///     Geometry.
         /// </summary>
         [Required]
-        public string Address { get; set; }
+        public string Geometry { get; set; }
```
Comment (tabeckers) [inline src/FunderMaps.Core/Entities/Building.cs:44]: Maybe do something with the geometry format in some way, indicating whether it's some form of JSON, hex encoding etc?

File: src/FunderMaps.Core/Entities/Building.cs (hunk 2 above)
Comment (tabeckers) [inline src/FunderMaps.Core/Entities/Building.cs:62]: Rename `buildingType` to `**B**uildingType`

**Review by tabeckers — CHANGES_REQUESTED** [review]
The major point for this PR is the inefficient Address x Building retrieval from our database. After some discussion this will be implemented in a future release (#287).

**Comment by tabeckers** [comment]
Also, just noticed the packages:

```Project `FunderMaps.Webservice` has the following updates to its packages
   [netcoreapp3.1]: 
   Top-level Package                         Requested   Resolved   Latest
   > Swashbuckle.AspNetCore                  5.6.0       5.6.0      5.6.1 
   > Swashbuckle.AspNetCore.Annotations      5.6.0       5.6.0      5.6.1 ```

File: src/FunderMaps.Core/Entities/Address.cs (hunk 1 above)
Comment (yorickdewid) [reply src/FunderMaps.Core/Entities/Address.cs:78]: No we don't, but there is a better solution proposed for a later time

File: src/FunderMaps.Core/Entities/Building.cs (hunk 4 above)
Comment (yorickdewid) [reply src/FunderMaps.Core/Entities/Building.cs:44]: Sure, I think it's best to create a new data type, but later

File: src/FunderMaps.Core/Entities/Building.cs (hunk 2 above)
Comment (yorickdewid) [reply src/FunderMaps.Core/Entities/Building.cs]: Fixed

**Review by tabeckers — APPROVED** [review]
Perfection

**Pull request merged**